File: Exercism/dart/beer-song/lib/beer_song.dart

class BeerSong {
  List<String> verse(int number) {
    switch (number) {
      case 0:
        return ['No more bottles of beer on the wall, no more bottles of beer.',
          'Go to the store and buy some more, 99 bottles of beer on the wall.'];

      case 1:
        return ['1 bottle of beer on the wall, 1 bottle of beer.',
        'Take it down and pass it around, no more bottles of beer on the wall.'];

      default:
        return ['$number bottles of beer on the wall, $number bottles of beer.',
          'Take one down and pass it around, ${number-1} bottle${(number == 2) ? "" : "s"} of beer on the wall.'];
    }
  }

  List<String> recite(int from, int count) {
    List<String> result = [];

    for (int i = 0 ; i < count ; i++) {
      result.addAll(verse(from-i));
      result.add('');
    }
    result.removeLast();

    return result;
  }
}

File: Exercism/python/beer-song/beer_song.py
def verse(x):
    if x == 0:
        return [
            "No more bottles of beer on the wall, no more bottles of beer.",
            "Go to the store and buy some more, 99 bottles of beer on the wall.",
        ]

    if x == 1:
        return [
            "1 bottle of beer on the wall, 1 bottle of beer.",
            "Take it down and pass it around, no more bottles of beer on the wall.",
        ]

    return [
        "{0} bottles of beer on the wall, {0} bottles of beer.".format(x),
        "Take one down and pass it around, {0} bottle{1} of beer on the wall."
        .format(x-1, "" if x == 2 else "s")
    ]


def recite(start, take=1):
    result = []

    while take > 0:
        result += verse(start)
        result.append("")
        start -= 1
        take -= 1

    return result[:-1]
